jobclient/python/jobs.py

```diff
@@ -0,0 +1,381 @@
+        :param status: The current status of the job, defaults to
+            Status.INITIALIZED
+        :type status: Status, optional
+        :param priority: The current priority of the job, defaults to 50
+        :type priority: int, optional
+        :param disable_mea_culpa_retries: If true, then the job will not issue
+            *mea culpa* retries, defaults to False
+        :type disable_mea_culpa_retries: bool, optional
+
+        Optional Parameters
+        -------------------
+        :param expected_runtime: This job's expected runtime, in seconds.
+        :type expected_runtime: Optional[int], optional
+        :param pool: The pool that this job should be deployed to.
+        :type pool: Optional[str], optional
+        :param instances: A list of instances for this job.
+        :type instances: Optional[List[Instance]], optional
+        :param env: A mapping of environment variables that should be set
+            before running this job.
+        :type env: Optional[Dict[str, str]], optional
+        :param uris: A list of URIs associated with this job.
+        :type uris: Optional[List[FetchableUri]], optional
+        :param container: Cotnainer description for this job.
+        :type container: Optional[dict], optional
+        :param labels: Labels associated with this job.
+        :type labels: Optional[Dict[str, str]], optional
+        :param constraints: Constraints for this job.
+        :type constraints: Optional[Set[Constraint]], optional
+        :param groups: Groups associated with this job.
+        :type groups: Optional[List[UUID]], optional
+        :param application: Application information for this job.
+        :type application: Optional[Application], optional
+        :param progress_output_file: Path to the file where this job will
+            output its progress.
+        :type progress_output_file: Optional[str], optional
+        :param progress_regex_string: Progress regex.
+        :type progress_regex_string: Optional[str], optional
+        :param user: User running this job.
+        :type user: Optional[str], optional
+        :param datasets: Datasets associated with this job.
+        :type datasets: Optional[list], optional
+        :param gpus: The number of GPUs to request from Cook.
+        :type gpus: Optional[float], optional
+        :param framework_id: Framework ID for this job.
+        :type framework_id: Optional[str], optional
+        :param ports: Ports for this job.
+        :type ports: Optional[int], optional
+        :param submit_time: The UNIX timestamp when this job was submitted.
+        :type submit_time: Optional[int], optional
+        :param retries_remaining: The number of retries remaining for this job.
+        :type retries_remaining: Optional[int], optional
+        """
+        self.command = command
+        self.mem = mem
+        self.cpus = cpus
+        self.uuid = uuid if uuid is not None else util.make_temporal_uuid()
```

Comment:
This is only for the query path, correct?

Reply:
Yup, will change!